neon_llm_fastchat/fastchat.py

```diff
@@ -78,3 +89,34 @@ def call_model(self, prompt):
         output_tokens = results[0].hypotheses[0]
         text = self.tokenizer.decode(self.tokenizer.convert_tokens_to_ids(output_tokens), spaces_between_special_tokens=False)
         return text
+
+    def call_score(self, prompt: str, targets: List[str]) -> List[List[float]]:
+        tokens = self.tokenize(prompt)
+        tokens_list = len(targets) * [tokens]
+
+        target_tokens_list = [self.tokenize(target) for target in targets]
+
+        results = self.model.score_batch(
+            source = tokens_list,
+            target = target_tokens_list,
+        )
+
+        log_probs_list = [result.log_probs for result in results]
+        return log_probs_list
+
+    @staticmethod
+    def compute_ppl(log_probs: List[float]) -> float:
+        "Calculates perplexity value based on https://en.wikipedia.org/wiki/Perplexity"
+        ppl = np.exp(-np.mean(log_probs))
+        return ppl
+
+    def ppl(self, question: str, answers: List[str]) -> List[float]:
+        question_prompt = self.assemble_prompt(question, [])
+        log_probs_list = self.call_score(question_prompt, answers)
+        ppl_list = [self.compute_ppl(log_probs) for log_probs in log_probs_list]
+        return ppl_list
+    
+    @staticmethod
+    def get_best(ppl_list: List[float]) -> int:
```

Comment:
Could you add docstrings and describe the parameters/return for these methods; I'm not sure what the args/return values represent for some of these